tests/e2e/encrypted.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+#
+# Copyright Confidential Containers Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+script_dir="$(dirname "$(readlink -f "$0")")"
+source "${script_dir}/lib.sh"
+
+export key_file="${key_file:-image_key}"
+export key_path="${key_path:-/default/image_key/nginx}"
+export trustee_repo="${trustee_repo:-https://github.com/confidential-containers/trustee.git}"
+export kbs_namespace="${kbs_namespace:-coco-tenant}"
+export aa_kbc="${aa_kbc:-cc_kbc}"
+export kbs_svc_name="${kbs_svc_name:-kbs}"
+export kbs_ingress_name="${kbs_ingress_name:-kbs}"
+export runtimeclass="${runtimeclass:-kata-qemu}"
+export username="${username:-}"
+export encrypted_image="${encrypted_image:-ghcr.io/$username/nginx:encrypted}"
+
+build_encryption_key() {
+	docker build -t unencrypted .
+	# Encryption key needs to be 32 byte sequence
+ 	head -c 32 /dev/urandom | openssl enc > "${key_file}"
+	# Encryption key needs to be provided as base 64 enconded string
+	export key_b64="$(base64 < ${key_file})"
+	export key_id="kbs://${key_path}"
+
+	# Image encryption logic
+	export guest_components_repo="https://github.com/confidential-containers/guest-components.git"
+	export guest_components_path="${script_dir}/guest-components"
+	if [ -d "${guest_components_path}" ]; then
+		echo "Guest components repo directory exists, removing it"
+		rm -rf "${guest_components_path}"
+	fi
+
+	git clone "${guest_components_repo}"
+
+	pushd "${guest_components_path}"
+  		export tag_name="coco-keyprovider"
+ 		docker build -t "${tag_name}" -f ./attestation-agent/docker/Dockerfile.keyprovider .
```

Comment:
guest-components [published an image](https://github.com/confidential-containers/guest-components/pkgs/container/staged-images%2Fcoco-keyprovider) of the keyprovider for the v0.10.0. There is a workflow to keep publishing that image for the next releases so I think we can consume the built one instead for build ourselves.